src/AzureExtension/Providers/RepositoryProvider.cs

```diff
@@ -100,129 +172,20 @@ IAsyncOperation<RepositoriesResult> IRepositoryProvider.GetRepositoriesAsync(IDe
             }).AsAsyncOperation();
         }
 
-        // Establish a connection to get all organizations.
-        // The library calls these organizations "accounts".
-        VssConnection accountConnection;
+        _azureHierarchy ??= new AzureRepositoryHierarchy(azureDeveloperId);
 
-        try
+        var server = string.Empty;
+        var organizations = _azureHierarchy.GetOrganizationsAsync().Result;

```

Comment:
I believe these do not use awaits because then the method would return to early and DevHome would move to the next screen.  I can try removing them and see what happens.

Reply:
Moved things over to using locks.  Got rid of the async as well.